### PR TITLE
Updated documentation for rvfi_trap

### PR DESCRIPTION
**Pull request by silabs-halfdan**
Updated documentation for rvfi_trap

### CODE REVIEW

File: docs/user_manual/source/rvfi.rst [hunk 1]
```diff
@@ -72,29 +72,58 @@ The trap signal indicates that a synchronous trap has ocurred and side-effects c
...
+  |                              |           |     |     |     +-------+--------+---------+-----------------+-----------------------------------------------------------------------+
+  |                              |           |     |     |     | 0xB   | X      | 0x0     | mcause, mepc    | ECALL executed from Machine mode                                      |
+  +------------------------------+-----------+-----+-----+-----+-------+--------+---------+-----------------+-----------------------------------------------------------------------+
+  | Instruction Bus Fault        | Exception | 1   | 1   | X   | 0x30  | X      | 0x0     | mcause, mepc    | OBI bus error on instruction fetch                                    |
+  +------------------------------+-----------+-----+-----+-----+-------+--------+---------+-----------------+-----------------------------------------------------------------------+
+  | Breakpoint to debug          | Debug     | 1   | X   | 1   | 0x0   | 0x1    | 0x0     | dpc, dcsr       | EBREAK from non-debug mode executed with  dcsr.ebreakm == 1           |
```
Comment (silabs-oysteink) [inline docs/user_manual/source/rvfi.rst:118]: [8:3] should be X'es for all debug causes I think?

File: docs/user_manual/source/rvfi.rst [hunk 2]
```diff
@@ -72,29 +72,58 @@ The trap signal indicates that a synchronous trap has ocurred and side-effects c
...
+  | Environment Call             | Exception | 1   | 1   | X   | 0x8   | X      | 0x0     | mcause, mepc    | ECALL executed from User mode                                         |
+  |                              |           |     |     |     +-------+--------+---------+-----------------+-----------------------------------------------------------------------+
+  |                              |           |     |     |     | 0xB   | X      | 0x0     | mcause, mepc    | ECALL executed from Machine mode                                      |
+  +------------------------------+-----------+-----+-----+-----+-------+--------+---------+-----------------+-----------------------------------------------------------------------+
+  | Instruction Bus Fault        | Exception | 1   | 1   | X   | 0x30  | X      | 0x0     | mcause, mepc    | OBI bus error on instruction fetch                                    |
+  +------------------------------+-----------+-----+-----+-----+-------+--------+---------+-----------------+-----------------------------------------------------------------------+
```
Comment (silabs-oysteink) [inline docs/user_manual/source/rvfi.rst:117]: Missing instruction parity/checksum fault

**Review by silabs-oysteink — CHANGES_REQUESTED** [review]

File: docs/user_manual/source/rvfi.rst (hunk 1 above)
Comment (silabs-halfdan) [reply docs/user_manual/source/rvfi.rst:118]: Yes, typo. Will fix.

File: docs/user_manual/source/rvfi.rst (hunk 2 above)
Comment (silabs-halfdan) [reply docs/user_manual/source/rvfi.rst:117]: Oversight, will add.

**Review by silabs-oysteink — APPROVED** [review]

**Pull request merged**